src/backend/distributed/commands/view.c

```diff
@@ -216,6 +216,42 @@ PreprocessDropViewStmt(Node *node, const char *queryString, ProcessUtilityContex
 }
 
 
+/*
+ * DropViewStmtObjectAddress returns object address for drop view statement.
+ */
+ObjectAddress
+DropViewStmtObjectAddress(Node *stmt, bool missing_ok)
+{
+	DropStmt *dropStmt = castNode(DropStmt, stmt);
+
+	ObjectAddress objectAddress = { 0 };
+
+	List *possiblyQualifiedViewName = NULL;
+	foreach_ptr(possiblyQualifiedViewName, dropStmt->objects)
+	{
+		char *viewName = NULL;
+		char *schemaName = NULL;
+		DeconstructQualifiedName(possiblyQualifiedViewName, &schemaName, &viewName);
+
+		if (schemaName == NULL)
```

Comment:
Why do we return invalid if `schemaName != NULL`? I cannot follow the need for calling `Deconstruct..` here again.

Wouldn't the following do the trick for us? 
```C

			RangeVar *viewRangeVar = makeRangeVarFromNameList(possiblyQualifiedViewName);
			Oid viewOid = RangeVarGetRelid(viewRangeVar, AccessExclusiveLock,
										  true);
if (OidIsValid(viewOid))
...
```

ps: not sure about `AccessExclusiveLock` at this moment